sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -1384,7 +1394,7 @@ def _generate_delete_blobs_options(
             try:
                 options = BlobClient._generic_delete_blob_options(  # pylint: disable=protected-access
                     snapshot=blob.get('snapshot'),
-                    version_id=blob.get('version_id'),
+                    version_id=kwargs.pop('version_id', None) or blob.get('version_id'),
```

Comment:
I don't think this change is needed? The overall `delete_blobs` method does not take a `version_id` since its present for each subrequest. This already handles pulling it from the properties for each subrequest, I don't think it will ever be in `kwargs`.

Same for `_generate_set_tiers_subrequest_options`